src/rust/multi.myreadingmanga/src/lib.rs

```diff
@@ -1,71 +1,103 @@
+	///
+	/// `wpsolr_sort`:
+	///
+	/// - `sort_by_relevancy_desc`: More relevant
+	/// - `sort_by_date_desc`: Newest
+	/// - `sort_by_date_asc`: Oldest
+	/// - `sort_by_random`: Random
+	///
+	/// `wpsolr_fq`:
+	///
+	/// - `index`: Start from `0`
+	/// - `filter_name` ➡️ Should be percent-encoded
+	Search(QueryParameters),
+
+	/// https://myreadingmanga.info/{manga_id}/
+	Manga(&'a str),
+
+	/// https://myreadingmanga.info/{manga_id}/{chapter_id}/
+	Chapter(&'a str, &'a str),
+}
 
 const DOMAIN: &str = "https://myreadingmanga.info";
+
+/// Safari on iOS 16.4
 const USER_AGENT: &str = "Mozilla/5.0 (iPhone; CPU iPhone OS 16_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.4 Mobile/15E148 Safari/604.1";
 
-fn get(url: String) -> Request {
-	Request::get(url)
-		.header("Referer", DOMAIN)
-		.header("User-Agent", USER_AGENT)
-}
+/// Sort by: \[More relevant, Newest, Oldest, Random\]
+const SORT: [&str; 4] = [
+	"sort_by_relevancy_desc",
+	"sort_by_date_desc",
+	"sort_by_date_asc",
+	"sort_by_random",
+];
 
 #[get_manga_list]
 fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
 	let manga_list_url = get_filtered_url(filters, page)?;
-	let manga_list_html = get(manga_list_url).html()?;
+	let manga_list_html = request_get(&manga_list_url).html()?;
 
-	let mut manga: Vec<Manga> = Vec::new();
-
-	for value in manga_list_html
-		.select("div.results-by-facets > div")
-		.array()
-	{
-		let manga_node = value.as_node()?;
+	let mut manga = Vec::<Manga>::new();
+	let manga_nodes = manga_list_html.select("div.results-by-facets > div");
+	for manga_value in manga_nodes.array() {
+		let manga_node = manga_value.as_node()?;
 		let title_node = manga_node.select("a");
 
-		let title = title_node.text().read();
+		let manga_title = title_node.text().read();
 
-		// Skip videos
-		// There are some exceptions
-		let is_video = !title.starts_with('[');
+		// ? Skip videos
+		// ! There are some exceptions
+		let is_video = !manga_title.starts_with('[');
 		if is_video {
 			continue;
 		}
 
-		let url = title_node.attr("href").read();
+		let manga_url = title_node.attr("href").read();
 
-		let id = url.replace(DOMAIN, "").replace('/', "");
+		let manga_id = manga_url.replace(DOMAIN, "").replace('/', "");
 
-		let cover = encode_uri(manga_node.select("img").attr("src").read());
+		let cover_url = manga_node
+			.select("img")
+			.attr("src")
+			.read()
+			.percent_encode(false);
 
-		let artist = match title.substring_before(']') {
-			Some(artists_str) => artists_str.replace('[', ""),
-			None => String::new(),
-		};
+		let artists_str = get_artists(&manga_title);
 
 		let description = manga_node.select("div.p_content").text().read();
 
 		manga.push(Manga {
-			id,
-			cover,
-			title,
-			author: artist.clone(),
-			artist,
+			id: manga_id,
+			cover: cover_url,
+			title: manga_title,
+			author: artists_str.clone(),
+			artist: artists_str,
 			description,
-			url,
+			url: manga_url,
```

Comment:
not really much reason to change these variable names, but doesn't really matter